libraries/libfc/src/variant.cpp

```diff
@@ -466,18 +466,26 @@ bool  variant::as_bool()const
    }
 }
 
+static std::string s_fc_to_string(double d)
+{
+   // +2 is required to ensure that the double is rounded correctly when read back in.  http://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html
+   std::stringstream ss;
+   ss << std::setprecision(std::numeric_limits<double>::digits10 + 2) << std::fixed << d;
+   return ss.str();
+}
+
 std::string variant::as_string()const
 {
    switch( get_type() )
    {
       case string_type:
           return **reinterpret_cast<const const_string_ptr*>(this);
       case double_type:
-          return to_string(*reinterpret_cast<const double*>(this));
+          return s_fc_to_string(*reinterpret_cast<const double*>(this));
```

Comment:
I don't believe this is used in consensus. Can you attempt to verify that. The difference in generated JSON might cause issues, but if we are planning on moving to a different JSON implementation that difference is likely to be hit anyway.

Reply:
> I don't believe this is used in consensus. Can you attempt to verify that. 

How can I check that conclusively?

Reply:
Thanks Kevin, will do!

Reply:
I verified that this one instance of `variant::as_string()` is never hit with type `double_type` when importing a snapshot, so I assume it means that it is safe to change to std::string.

> Visual code inspection.

I don't quite understand what would cause a change in consensus. I would have thought that only changes in the wasm generated code for contracts or the WASM execution engine could trigger consensus issues, which shows how little I know. So I don't know what to check.